test_public/modules/soumetsHomologation.spec.mjs

```diff
@@ -0,0 +1,100 @@
+      global.$ = jquery(dom.window);
+
+      global.window = { location: '/' };
+    });
+
+    it('affiche la modale quand le seuil de criticité est critique', (done) => {
+      let modaleAffichee = false;
+      const adaptateurAjax = {
+        get: () => Promise.resolve({ data: { seuilCriticite: 'critique' } }),
+      };
+      $('*').on('afficheModale', (evenement) => {
+        evenement.preventDefault();
+        modaleAffichee = true;
+      });
+
+      Promise.resolve(initialiseComportementFormulaire('.formulaire', '#bouton', { adaptateurAjax }))
+        .then(() => Promise.resolve($('.formulaire').trigger('submit')))
+        .then(() => {
+          try {
+            expect(modaleAffichee).to.be(true);
+            done();
+          } catch (erreur) {
+            done(erreur);
+          }
+        });
+    });
+
+    describe("quand le seuil de criticité n'est pas critique", () => {
+      let ajaxRequete;
+      let adaptateurAjax;
+
+      beforeEach(() => {
+        adaptateurAjax = (requete) => {
+          ajaxRequete = requete;
+          return Promise.resolve({ data: { idHomologation: '123' } });
+        };
+        adaptateurAjax.get = () => Promise.resolve({ data: { seuilCriticite: 'normal' } });
+      });
+
+      it("Écris une nouvelle homologation si elle n'est pas identifiée", (done) => {
+        Promise.resolve(initialiseComportementFormulaire('.formulaire', '#bouton', { adaptateurAjax }))
+          .then(() => Promise.resolve($('.formulaire').trigger('submit')))
+          .then(() => {
+            try {
+              expect(ajaxRequete.method).to.equal('post');
+              expect(ajaxRequete.url).to.equal('/api/homologation');
+              expect(ajaxRequete.data['champs-1']).to.equal('valeur 1');
+              done();
+            } catch (erreur) {
+              done(erreur);
+            }
+          });
+      });
+
+      it("Mets à jour l'homologation si elle est identifiée", (done) => {
+        $('.bouton').attr('identifiant', '12345');
+
+        Promise.resolve(initialiseComportementFormulaire('.formulaire', '.bouton', { adaptateurAjax }))
+          .then(() => Promise.resolve($('.formulaire').trigger('submit')))
+          .then(() => {
+            try {
+              expect(ajaxRequete.method).to.equal('put');
+              expect(ajaxRequete.url).to.equal('/api/homologation/12345');
+              expect(ajaxRequete.data['champs-1']).to.equal('valeur 1');
+              done();
+            } catch (erreur) {
+              done(erreur);
+            }
+          });
+      });
+
+      it("Renvoie vers la synthèse de l'homologation", (done) => {
+        Promise.resolve(initialiseComportementFormulaire('.formulaire', '.bouton', { adaptateurAjax }))
+          .then(() => Promise.resolve(($('.formulaire').trigger('submit'))))
+          .then(() => {
+            try {
+              expect(global.window.location).to.equal('/homologation/123');
```

Comment:
Je ne suis pas sûr qu'on ait besoin ici d'écrire explicitement `global`. `expect(window.location).to.equal(…)` ne suffirait-il pas ?

Reply:
oui en effet